rasa/graph_components/validators/config_validator.py

```diff
@@ -0,0 +1,514 @@
+                )
+
+        if training_data.regex_features and self._component_types.isdisjoint(
+            [RegexFeaturizerGraphComponent, RegexEntityExtractorGraphComponent],
+        ):
+            rasa.shared.utils.io.raise_warning(
+                f"You have defined training data with regexes, but "
+                f"your NLU configuration does not include a 'RegexFeaturizer' "
+                f" or a "
+                f"'RegexEntityExtractor'. To use regexes, include either a "
+                f"'{RegexFeaturizerGraphComponent.__name__}' or a "
+                f"'{RegexEntityExtractorGraphComponent.__name__}' "
+                f"in your configuration.",
+                docs=DOCS_URL_COMPONENTS,
+            )
+
+        if training_data.lookup_tables and self._component_types.isdisjoint(
+            [RegexFeaturizerGraphComponent, RegexEntityExtractorGraphComponent],
+        ):
+            rasa.shared.utils.io.raise_warning(
+                f"You have defined training data consisting of lookup tables, but "
+                f"your NLU configuration does not include a featurizer "
+                f"or an entity extractor using the lookup table."
+                f"To use the lookup tables, include either a "
+                f"'{RegexFeaturizerGraphComponent.__name__}' "
+                f"or a '{RegexEntityExtractorGraphComponent.__name__}' "
+                f"in your configuration.",
+                docs=DOCS_URL_COMPONENTS,
+            )
+
+        if training_data.lookup_tables:
+
+            if self._component_types.isdisjoint(
+                [CRFEntityExtractorGraphComponent, DIETClassifierGraphComponent]
+            ):
+                rasa.shared.utils.io.raise_warning(
+                    f"You have defined training data consisting of lookup tables, but "
+                    f"your NLU configuration does not include any components "
+                    f"that uses the features created from the lookup table. "
+                    f"To make use of the features that are created with the "
+                    f"help of the lookup tables, "
+                    f"add a '{DIETClassifierGraphComponent.__name__}' or a "
+                    f"'{CRFEntityExtractorGraphComponent.__name__}' "
+                    f"with the 'pattern' feature "
+                    f"to your configuration.",
+                    docs=DOCS_URL_COMPONENTS,
+                )
+
+            elif CRFEntityExtractorGraphComponent in self._component_types:
+
+                crf_schema_nodes = [
+                    schema_node
+                    for schema_node in self._graph_schema.nodes.values()
+                    if schema_node.uses == CRFEntityExtractorGraphComponent
+                ]
+
+                has_pattern_feature = False
+                for crf in crf_schema_nodes:
+                    crf_features = crf.config.get("features", [])
+                    has_pattern_feature = "pattern" in itertools.chain(*crf_features)
```

Comment:
optional: Would it be worth introducing a constant for `pattern`?

Reply:
_theoratically_ there could be multiple CRFEntityExtractors and then it's enough if only one of them uses `pattern`, right? In that case we should probably do something like this?
```suggestion
                    has_pattern_feature = has_pattern_feature or "pattern" in itertools.chain(*crf_features)
```

Reply:
> optional: Would it be worth introducing a constant for pattern?
agree :) same for "features"

Reply:
> theoratically there could be multiple CRFEntityExtractors and then it's enough if only one of them uses pattern, right? In that case we should probably do something like this?

nice observation :) will add that